platforms/ide/ide-native/src/main/java/org/gradle/ide/visualstudio/plugins/VisualStudioPlugin.java

```diff
@@ -187,16 +187,19 @@ private void configureCleanTask() {
         });
     }
 
+    @SuppressWarnings("deprecation")
     private Task createSolutionTask(VisualStudioSolution solution) {
         return project.getTasks().create(solution.getName() + "VisualStudioSolution", GenerateSolutionFileTask.class, solution);
     }
 
+    @SuppressWarnings("deprecation")
     private Task createProjectsFileTask(VisualStudioProject vsProject) {
         GenerateProjectFileTask task = project.getTasks().create(vsProject.getName() + "VisualStudioProject", GenerateProjectFileTask.class, vsProject);
         task.initGradleCommand();
         return task;
     }
 
+    @SuppressWarnings("deprecation")
     private Task createFiltersFileTask(VisualStudioProject vsProject) {
         return project.getTasks().create(vsProject.getName() + "VisualStudioFilters", GenerateFiltersFileTask.class, vsProject);
```

Comment:
💭 By moving this `.java` file from `src/main/groovy` to `src/main/java` these invocations of `tasks.create(...)` are now flagged by `-Werror` of `javac`.